src/core/Table/filters/NumberRangeFilter/NumberRangeFilter.tsx

```diff
@@ -0,0 +1,86 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../../../utils/hooks/useTheme';
+import {
+  FilterButtonBar,
+  FilterButtonBarTranslation,
+} from '../FilterButtonBar';
+import { BaseFilter } from '../BaseFilter';
+import { TableFilterProps } from '../types';
+import '@itwin/itwinui-css/css/table.css';
+import { LabeledInput } from '../../../LabeledInput';
+
+export type NumberRangeTranslation = {
+  from: string;
+  to: string;
+};
+
+const defaultStrings: NumberRangeTranslation = {
+  from: 'From',
+  to: 'To',
+};
+
+export type NumberRangeFilterProps<
+  T extends Record<string, unknown>
+> = TableFilterProps<T> & {
+  translatedLabels?: NumberRangeTranslation & FilterButtonBarTranslation;
+};
+
+export const NumberRangeFilter = <T extends Record<string, unknown>>(
+  props: NumberRangeFilterProps<T>,
+) => {
+  const { column, translatedLabels, setFilter, clearFilter } = props;
+
+  useTheme();
+
+  const translatedStrings = { ...defaultStrings, ...translatedLabels };
+
+  const [from, setFrom] = React.useState<string | undefined>(
+    column.filterValue?.[0] ?? '',
+  );
+  const [to, setTo] = React.useState<string | undefined>(
+    column.filterValue?.[1] ?? '',
+  );
+
+  const parseInputValue = (value: string | undefined) => {
+    return !value || isNaN(Number(value)) ? undefined : Number(value);
```

Comment:
Weird logic. Maybe we should allow user to also provide their own parser, similar to date.

Also got me thinking it could be useful for float type values 🤔

Reply:
I need to convert string to number `Number(undefined) === 0` so I have to check whether `value` is defined - it won't allow `''`, `undefined`.

Reply:
and what about float values?

Reply:
It support float.